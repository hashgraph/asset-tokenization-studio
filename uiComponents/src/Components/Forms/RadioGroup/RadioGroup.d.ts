import type { RadioGroupProps as ChakraRadioGroupProps } from "@chakra-ui/radio";
import type { ComponentWithAs } from "@chakra-ui/system";
import type { BaseSingleStyleConfiguration } from "@Theme/Components/BaseSingleStyleConfiguration";
export interface RadioGroupThemeConfiguration extends BaseSingleStyleConfiguration {
}
export interface RadioGroupProps extends ChakraRadioGroupProps {
}
export declare const RadioGroup: ComponentWithAs<"div", RadioGroupProps>;
