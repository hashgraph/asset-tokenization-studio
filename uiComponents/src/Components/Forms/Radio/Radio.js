var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { useFormControlContext as useChakraFormControlContext } from "@chakra-ui/form-control";
import { Radio as ChakraRadio } from "@chakra-ui/radio";
import React from "react";
import { forwardRef } from "@chakra-ui/system";
export var radioPartsList = [
    "container",
    "label",
    "control",
];
export var Radio = forwardRef(function (_a, ref) {
    var name = _a.name, isInvalid = _a.isInvalid, label = _a.label, children = _a.children, size = _a.size, variant = _a.variant, isDisabled = _a.isDisabled, props = __rest(_a, ["name", "isInvalid", "label", "children", "size", "variant", "isDisabled"]);
    var formControl = useChakraFormControlContext() || {};
    return (React.createElement(ChakraRadio, __assign({ ref: ref, isInvalid: isInvalid !== null && isInvalid !== void 0 ? isInvalid : formControl.isInvalid, name: name, size: size, variant: variant, isDisabled: isDisabled }, props), children));
});
