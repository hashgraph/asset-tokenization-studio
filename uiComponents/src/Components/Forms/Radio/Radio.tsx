import type { radioAnatomy as ChakraRadioParts } from "@chakra-ui/anatomy";
import { useFormControlContext as useChakraFormControlContext } from "@chakra-ui/form-control";
import type { RadioProps as ChakraRadioProps } from "@chakra-ui/radio";
import { Radio as ChakraRadio } from "@chakra-ui/radio";
import React from "react";
import type { BaseMultiStyleConfiguration } from "@/Theme/Components/BaseMultiStyleConfiguration";
import type { ComponentWithAs } from "@chakra-ui/system";
import { forwardRef } from "@chakra-ui/system";

export const radioPartsList: typeof ChakraRadioParts.keys = [
  "container",
  "label",
  "control",
];

type Parts = typeof radioPartsList;

export interface RadioThemeConfiguration
  extends BaseMultiStyleConfiguration<Parts> {}

export interface RadioProps extends Omit<ChakraRadioProps, "colorScheme"> {
  label?: string;
}

export const Radio: ComponentWithAs<"input", RadioProps> = forwardRef<
  RadioProps,
  "input"
>(
  (
    {
      name,
      isInvalid,
      label,
      children,
      size,
      variant,
      isDisabled,
      ...props
    }: RadioProps,
    ref
  ) => {
    const formControl = useChakraFormControlContext() || {};

    return (
      <ChakraRadio
        ref={ref}
        isInvalid={isInvalid ?? formControl.isInvalid}
        name={name}
        size={size}
        variant={variant}
        isDisabled={isDisabled}
        {...props}
      >
        {children}
      </ChakraRadio>
    );
  }
);
