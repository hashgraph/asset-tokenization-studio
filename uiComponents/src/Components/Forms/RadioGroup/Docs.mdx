import {
  Story,
  Meta,
  Title,
  ArgTypes,
  Primary,
  Stories,
  Source,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as RadioGroupStories from "./RadioGroup.stories";

<Meta of={RadioGroupStories} />

<Title />

<a
  href="https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1874%3A15429&t=GPCIOt6k4o0cCWIe-4"
  target="\_blank"
>
  View figma design
</a>

<Primary />

## How to set up

RadioGroup is built with RadioGroup of Chakra UI

## Props

<ArgTypes of={RadioGroupStories} />

## Stories

<Stories title="" />
