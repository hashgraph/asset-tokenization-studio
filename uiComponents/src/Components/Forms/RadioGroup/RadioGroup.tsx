import type { RadioGroupProps as ChakraRadioGroupProps } from "@chakra-ui/radio";
import { RadioGroup as ChakraRadioGroup } from "@chakra-ui/radio";
import type { ComponentWithAs } from "@chakra-ui/system";
import { forwardRef } from "@chakra-ui/system";
import type { BaseSingleStyleConfiguration } from "@Theme/Components/BaseSingleStyleConfiguration";
import React from "react";

export interface RadioGroupThemeConfiguration
  extends BaseSingleStyleConfiguration {}

export interface RadioGroupProps extends ChakraRadioGroupProps {}

export const RadioGroup: ComponentWithAs<"div", RadioGroupProps> = forwardRef<
  RadioGroupProps,
  "div"
>(({ name, ...props }: RadioGroupProps, ref) => {
  return <ChakraRadioGroup ref={ref} name={name} {...props} />;
});
