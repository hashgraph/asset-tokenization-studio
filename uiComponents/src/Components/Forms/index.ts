export * from "./AddAreaButton";
export * from "./CalendarInput";
export * from "./Checkbox";
export * from "./FileInput";
export * from "./Input";
export * from "./Radio";
export * from "./RadioGroup";
export * from "./Select";
export * from "./Slider";
export * from "./Textarea";
export * from "./Toggle";
export * from "./Controllers";
