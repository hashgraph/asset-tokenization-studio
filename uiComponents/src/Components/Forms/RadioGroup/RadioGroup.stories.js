var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { Stack as ChakraStack } from "@chakra-ui/react";
import React from "react";
import { Radio } from "../Radio/Radio";
import { RadioGroup } from "./RadioGroup";
import { linkTo } from "@storybook/addon-links";
import { Button } from "@Components/Interaction/Button";
var meta = {
    title: "Design System/Forms/RadioGroup",
    component: RadioGroup,
    args: {
        children: (React.createElement(ChakraStack, { align: "flex-start" },
            React.createElement(Radio, { value: "check1" }, "Check1"),
            React.createElement(Radio, { value: "check2" }, "Check2"),
            React.createElement(Radio, { value: "check3" }, "Check3")))
    },
    argTypes: {
        isDisabled: { control: { type: "boolean" } },
        defaultValue: { control: { type: "text" } },
        name: { control: { type: "text" } }
    },
    parameters: {
        design: {
            type: "figma",
            url: "https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1874%3A15429&t=GPCIOt6k4o0cCWIe-4"
        },
        docs: {}
    }
};
export default meta;
var Template = function (args) { return React.createElement(RadioGroup, __assign({}, args)); };
export var DefaultValue = Template.bind({});
DefaultValue.args = {
    defaultValue: "check2"
};
export var IsDisabled = Template.bind({});
IsDisabled.args = {
    isDisabled: true
};
export var ControlledRadioGroup = function () { return (React.createElement(Button, { onClick: linkTo("Design System/Forms/Controllers/RadioGroupController") }, "Check out the RadioGroupController component Stories")); };
