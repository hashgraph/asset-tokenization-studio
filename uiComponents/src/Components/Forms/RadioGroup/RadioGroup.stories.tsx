import { Stack as ChakraStack } from "@chakra-ui/react";
import type { Meta, StoryFn } from "@storybook/react";
import React from "react";

import { Radio } from "../Radio/Radio";
import { RadioGroup } from "./RadioGroup";
import { linkTo } from "@storybook/addon-links";
import { Button } from "@Components/Interaction/Button";

const meta = {
  title: "Design System/Forms/RadioGroup",
  component: RadioGroup,
  args: {
    children: (
      <ChakraStack align="flex-start">
        <Radio value="check1">Check1</Radio>
        <Radio value="check2">Check2</Radio>
        <Radio value="check3">Check3</Radio>
      </ChakraStack>
    ),
  },
  argTypes: {
    isDisabled: { control: { type: "boolean" } },
    defaultValue: { control: { type: "text" } },
    name: { control: { type: "text" } },
  },
  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1874%3A15429&t=GPCIOt6k4o0cCWIe-4",
    },
    docs: {},
  },
} as Meta<typeof RadioGroup>;
export default meta;

const Template: StoryFn<typeof RadioGroup> = (args) => <RadioGroup {...args} />;

export const DefaultValue = Template.bind({});
DefaultValue.args = {
  defaultValue: "check2",
};

export const IsDisabled = Template.bind({});
IsDisabled.args = {
  isDisabled: true,
};

export const ControlledRadioGroup = () => (
  <Button
    onClick={linkTo("Design System/Forms/Controllers/RadioGroupController")}
  >
    Check out the RadioGroupController component Stories
  </Button>
);
