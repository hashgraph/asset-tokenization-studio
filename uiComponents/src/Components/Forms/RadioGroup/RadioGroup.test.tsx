import { render } from "@/test-utils";
import { Radio } from "@Components/Forms/Radio";
import React from "react";
import type { RadioGroupProps } from "./RadioGroup";
import { RadioGroup } from "./RadioGroup";

describe(`<RadioGroup />`, () => {
  const childrenProp = {
    children: (
      <>
        <Radio value="check1">Check1</Radio>
        <Radio value="check2">Check2</Radio>
        <Radio value="check3">Check3</Radio>
      </>
    ),
  };
  const isDisabledProp = { isDisabled: true };
  const defaultValueProp = { defaultValue: "check2" };

  const factoryComponent = (props?: Partial<RadioGroupProps>) =>
    render(<RadioGroup name="test-group" {...childrenProp} {...props} />);

  test("renders correctly", () => {
    const component = factoryComponent();

    expect(component.asFragment()).toMatchSnapshot();
  });

  test("shows correctly when is disabled", async () => {
    const component = factoryComponent(isDisabledProp);

    expect(component.asFragment()).toMatchSnapshot("Using as disabled");
  });

  test("shows correctly when sets value by default", () => {
    const component = factoryComponent(defaultValueProp);

    expect(component.asFragment()).toMatchSnapshot("Using as default value");
  });

  test("shows correctly when is checked and disabled", () => {
    const component = factoryComponent({
      ...defaultValueProp,
      ...isDisabledProp,
    });

    expect(component.asFragment()).toMatchSnapshot(
      "Using as default checked and disabled"
    );
  });
});
