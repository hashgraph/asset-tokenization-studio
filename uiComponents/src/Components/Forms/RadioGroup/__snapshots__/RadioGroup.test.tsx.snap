// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="chakra-radio-group css-0"
    role="radiogroup"
  >
    <label
      class="chakra-radio css-1c75yry"
    >
      <input
        class="chakra-radio__input"
        id="radio-:r1:"
        name="test-group"
        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
        type="radio"
        value="check1"
      />
      <span
        aria-hidden="true"
        class="chakra-radio__control css-1oe91ey"
      />
      <span
        class="chakra-radio__label css-1ngpyuv"
      >
        Check1
      </span>
    </label>
    <label
      class="chakra-radio css-1c75yry"
    >
      <input
        class="chakra-radio__input"
        id="radio-:r2:"
        name="test-group"
        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
        type="radio"
        value="check2"
      />
      <span
        aria-hidden="true"
        class="chakra-radio__control css-1oe91ey"
      />
      <span
        class="chakra-radio__label css-1ngpyuv"
      >
        Check2
      </span>
    </label>
    <label
      class="chakra-radio css-1c75yry"
    >
      <input
        class="chakra-radio__input"
        id="radio-:r3:"
        name="test-group"
        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
        type="radio"
        value="check3"
      />
      <span
        aria-hidden="true"
        class="chakra-radio__control css-1oe91ey"
      />
      <span
        class="chakra-radio__label css-1ngpyuv"
      >
        Check3
      </span>
    </label>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<RadioGroup /> shows correctly when is checked and disabled: Using as default checked and disabled 1`] = `
<DocumentFragment>
  <div
    class="chakra-radio-group css-0"
    role="radiogroup"
  >
    <label
      class="chakra-radio css-1c75yry"
      data-disabled=""
    >
      <input
        aria-disabled="true"
        class="chakra-radio__input"
        disabled=""
        id="radio-:rd:"
        name="test-group"
        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
        type="radio"
        value="check1"
      />
      <span
        aria-hidden="true"
        class="chakra-radio__control css-1oe91ey"
        data-disabled=""
      />
      <span
        class="chakra-radio__label css-1ngpyuv"
        data-disabled=""
      >
        Check1
      </span>
    </label>
    <label
      class="chakra-radio css-1c75yry"
      data-checked=""
      data-disabled=""
    >
      <input
        aria-disabled="true"
        checked=""
        class="chakra-radio__input"
        disabled=""
        id="radio-:re:"
        name="test-group"
        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
        type="radio"
        value="check2"
      />
      <span
        aria-hidden="true"
        class="chakra-radio__control css-1oe91ey"
        data-checked=""
        data-disabled=""
      />
      <span
        class="chakra-radio__label css-1ngpyuv"
        data-checked=""
        data-disabled=""
      >
        Check2
      </span>
    </label>
    <label
      class="chakra-radio css-1c75yry"
      data-disabled=""
    >
      <input
        aria-disabled="true"
        class="chakra-radio__input"
        disabled=""
        id="radio-:rf:"
        name="test-group"
        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
        type="radio"
        value="check3"
      />
      <span
        aria-hidden="true"
        class="chakra-radio__control css-1oe91ey"
        data-disabled=""
      />
      <span
        class="chakra-radio__label css-1ngpyuv"
        data-disabled=""
      >
        Check3
      </span>
    </label>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<RadioGroup /> shows correctly when is disabled: Using as disabled 1`] = `
<DocumentFragment>
  <div
    class="chakra-radio-group css-0"
    role="radiogroup"
  >
    <label
      class="chakra-radio css-1c75yry"
      data-disabled=""
    >
      <input
        aria-disabled="true"
        class="chakra-radio__input"
        disabled=""
        id="radio-:r5:"
        name="test-group"
        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
        type="radio"
        value="check1"
      />
      <span
        aria-hidden="true"
        class="chakra-radio__control css-1oe91ey"
        data-disabled=""
      />
      <span
        class="chakra-radio__label css-1ngpyuv"
        data-disabled=""
      >
        Check1
      </span>
    </label>
    <label
      class="chakra-radio css-1c75yry"
      data-disabled=""
    >
      <input
        aria-disabled="true"
        class="chakra-radio__input"
        disabled=""
        id="radio-:r6:"
        name="test-group"
        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
        type="radio"
        value="check2"
      />
      <span
        aria-hidden="true"
        class="chakra-radio__control css-1oe91ey"
        data-disabled=""
      />
      <span
        class="chakra-radio__label css-1ngpyuv"
        data-disabled=""
      >
        Check2
      </span>
    </label>
    <label
      class="chakra-radio css-1c75yry"
      data-disabled=""
    >
      <input
        aria-disabled="true"
        class="chakra-radio__input"
        disabled=""
        id="radio-:r7:"
        name="test-group"
        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
        type="radio"
        value="check3"
      />
      <span
        aria-hidden="true"
        class="chakra-radio__control css-1oe91ey"
        data-disabled=""
      />
      <span
        class="chakra-radio__label css-1ngpyuv"
        data-disabled=""
      >
        Check3
      </span>
    </label>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<RadioGroup /> shows correctly when sets value by default: Using as default value 1`] = `
<DocumentFragment>
  <div
    class="chakra-radio-group css-0"
    role="radiogroup"
  >
    <label
      class="chakra-radio css-1c75yry"
    >
      <input
        class="chakra-radio__input"
        id="radio-:r9:"
        name="test-group"
        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
        type="radio"
        value="check1"
      />
      <span
        aria-hidden="true"
        class="chakra-radio__control css-1oe91ey"
      />
      <span
        class="chakra-radio__label css-1ngpyuv"
      >
        Check1
      </span>
    </label>
    <label
      class="chakra-radio css-1c75yry"
      data-checked=""
    >
      <input
        checked=""
        class="chakra-radio__input"
        id="radio-:ra:"
        name="test-group"
        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
        type="radio"
        value="check2"
      />
      <span
        aria-hidden="true"
        class="chakra-radio__control css-1oe91ey"
        data-checked=""
      />
      <span
        class="chakra-radio__label css-1ngpyuv"
        data-checked=""
      >
        Check2
      </span>
    </label>
    <label
      class="chakra-radio css-1c75yry"
    >
      <input
        class="chakra-radio__input"
        id="radio-:rb:"
        name="test-group"
        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
        type="radio"
        value="check3"
      />
      <span
        aria-hidden="true"
        class="chakra-radio__control css-1oe91ey"
      />
      <span
        class="chakra-radio__label css-1ngpyuv"
      >
        Check3
      </span>
    </label>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
